In [1]:
import pandas as pd

In [54]:
dataset=pd.read_csv("Placement.csv")

In [55]:
dataset

sl_no gender  ssc_p    ssc_b  hsc_p    hsc_b     hsc_s  degree_p  \
0        1      M  67.00   Others  91.00   Others  Commerce     58.00   
1        2      M  79.33  Central  78.33   Others   Science     77.48   
2        3      M  65.00  Central  68.00  Central      Arts     64.00   
3        4      M  56.00  Central  52.00  Central   Science     52.00   
4        5      M  85.80  Central  73.60  Central  Commerce     73.30   
..     ...    ...    ...      ...    ...      ...       ...       ...   
210    211      M  80.60   Others  82.00   Others  Commerce     77.60   
211    212      M  58.00   Others  60.00   Others   Science     72.00   
212    213      M  67.00   Others  67.00   Others  Commerce     73.00   
213    214      F  74.00   Others  66.00   Others  Commerce     58.00   
214    215      M  62.00  Central  58.00   Others   Science     53.00   

      degree_t workex  etest_p specialisation  mba_p      status    salary  
0     Sci&Tech     No     55.0         Mkt&HR  58.80      Placed  270000.0  
1     Sci&Tech    Yes     86.5        Mkt&Fin  66.28      Placed  200000.0  
2    Comm&Mgmt     No     75.0        Mkt&Fin  57.80      Placed  250000.0  
3     Sci&Tech     No     66.0         Mkt&HR  59.43  Not Placed       NaN  
4    Comm&Mgmt     No     96.8        Mkt&Fin  55.50      Placed  425000.0  
..         ...    ...      ...            ...    ...         ...       ...  
210  Comm&Mgmt     No     91.0        Mkt&Fin  74.49      Placed  400000.0  
211   Sci&Tech     No     74.0        Mkt&Fin  53.62      Placed  275000.0  
212  Comm&Mgmt    Yes     59.0        Mkt&Fin  69.72      Placed  295000.0  
213  Comm&Mgmt     No     70.0         Mkt&HR  60.23      Placed  204000.0  
214  Comm&Mgmt     No     89.0         Mkt&HR  60.22  Not Placed       NaN  

[215 rows x 15 columns]

In [56]:
dataset.isna().sum()

sl_no              0
gender             0
ssc_p              0
ssc_b              0
hsc_p              0
hsc_b              0
hsc_s              0
degree_p           0
degree_t           0
workex             0
etest_p            0
specialisation     0
mba_p              0
status             0
salary            67
dtype: int64

In [57]:
dataset.salary.min()

200000.0

# 1)Replace the NaN values with correct value. And justify why you have chosen the same.


# In this dataset only Salary colunm has NaN value it depends upon the status column. Their status shown as Placed,they get salary and status shown as not Placed,it shows NaN value.  We shouldn't change the base information of data so we can replace Nan value with 0,it will not affect the basic information of the data.

In [5]:
dataset["salary"].fillna(0,inplace=True)

In [6]:
dataset

sl_no gender  ssc_p    ssc_b  hsc_p    hsc_b     hsc_s  degree_p  \
0        1      M  67.00   Others  91.00   Others  Commerce     58.00   
1        2      M  79.33  Central  78.33   Others   Science     77.48   
2        3      M  65.00  Central  68.00  Central      Arts     64.00   
3        4      M  56.00  Central  52.00  Central   Science     52.00   
4        5      M  85.80  Central  73.60  Central  Commerce     73.30   
..     ...    ...    ...      ...    ...      ...       ...       ...   
210    211      M  80.60   Others  82.00   Others  Commerce     77.60   
211    212      M  58.00   Others  60.00   Others   Science     72.00   
212    213      M  67.00   Others  67.00   Others  Commerce     73.00   
213    214      F  74.00   Others  66.00   Others  Commerce     58.00   
214    215      M  62.00  Central  58.00   Others   Science     53.00   

      degree_t workex  etest_p specialisation  mba_p      status    salary  
0     Sci&Tech     No     55.0         Mkt&HR  58.80      Placed  270000.0  
1     Sci&Tech    Yes     86.5        Mkt&Fin  66.28      Placed  200000.0  
2    Comm&Mgmt     No     75.0        Mkt&Fin  57.80      Placed  250000.0  
3     Sci&Tech     No     66.0         Mkt&HR  59.43  Not Placed       0.0  
4    Comm&Mgmt     No     96.8        Mkt&Fin  55.50      Placed  425000.0  
..         ...    ...      ...            ...    ...         ...       ...  
210  Comm&Mgmt     No     91.0        Mkt&Fin  74.49      Placed  400000.0  
211   Sci&Tech     No     74.0        Mkt&Fin  53.62      Placed  275000.0  
212  Comm&Mgmt    Yes     59.0        Mkt&Fin  69.72      Placed  295000.0  
213  Comm&Mgmt     No     70.0         Mkt&HR  60.23      Placed  204000.0  
214  Comm&Mgmt     No     89.0         Mkt&HR  60.22  Not Placed       0.0  

[215 rows x 15 columns]

In [7]:
dataset.isna().sum()

sl_no             0
gender            0
ssc_p             0
ssc_b             0
hsc_p             0
hsc_b             0
hsc_s             0
degree_p          0
degree_t          0
workex            0
etest_p           0
specialisation    0
mba_p             0
status            0
salary            0
dtype: int64

dir(dataset)

In [8]:
freqTable=pd.DataFrame(columns=["unique_values","Frequency","Relative_Frequency","Cumsum"])

In [9]:
freqTable

Empty DataFrame
Columns: [unique_values, Frequency, Relative_Frequency, Cumsum]
Index: []

In [10]:
freqTable["unique_values"]=dataset["status"].value_counts().index
freqTable["Frequency"]=dataset["status"].value_counts().values
freqTable["Relative_Frequency"]=(freqTable["Frequency"]/215)
freqTable["Cumsum"]=freqTable["Relative_Frequency"].cumsum()

In [11]:
freqTable

unique_values  Frequency  Relative_Frequency    Cumsum
0        Placed        148            0.688372  0.688372
1    Not Placed         67            0.311628  1.000000

# 2.How many of them are not placed?

# 67 persons not-placed

In [12]:
dataset[dataset["gender"]=="M"]["status"]

0          Placed
1          Placed
2          Placed
3      Not Placed
4          Placed
          ...    
209        Placed
210        Placed
211        Placed
212        Placed
214    Not Placed
Name: status, Length: 139, dtype: object

In [13]:
freqTable=pd.DataFrame(columns=["unique_values","Frequency"])
freqTable["unique_values"]=dataset[dataset["gender"]=="M"]["status"].value_counts().index
freqTable["Frequency"]=dataset[dataset["gender"]=="M"]["status"].value_counts().values

In [14]:
freqTable

unique_values  Frequency
0        Placed        100
1    Not Placed         39

In [15]:
dataset[dataset["gender"]=="F"]["status"]

6      Not Placed
12     Not Placed
13         Placed
15         Placed
17     Not Placed
          ...    
197        Placed
198    Not Placed
204        Placed
208    Not Placed
213        Placed
Name: status, Length: 76, dtype: object

In [16]:
freqTable=pd.DataFrame(columns=["unique_values","Frequency"])
freqTable["unique_values"]=dataset[dataset["gender"]=="F"]["status"].value_counts().index
freqTable["Frequency"]=dataset[dataset["gender"]=="F"]["status"].value_counts().values

In [17]:
freqTable

unique_values  Frequency
0        Placed         48
1    Not Placed         28

In [18]:
from scipy.stats import ttest_rel
dataset=dataset.dropna()
st = dataset[dataset['status']=='Not Placed']['ssc_p']
st1 = dataset[dataset['status']=='Not Placed']['hsc_p']
ttest_rel(st,st1)

Ttest_relResult(statistic=-0.6581198043994656, pvalue=0.5127502684992262)

In [19]:
from scipy.stats import ttest_rel
dataset=dataset.dropna()
st2 = dataset[dataset['status']=='Not Placed']['degree_p']
st3 = dataset[dataset['status']=='Not Placed']['etest_p']
ttest_rel(st2,st3)

Ttest_relResult(statistic=-5.366918080573216, pvalue=1.109973762243531e-06)

In [20]:
from scipy.stats import ttest_rel
dataset=dataset.dropna()
st3 = dataset[dataset['status']=='Not Placed']['etest_p']
st4 = dataset[dataset['status']=='Not Placed']['mba_p']
ttest_rel(st3,st4)

Ttest_relResult(statistic=4.969125724283872, pvalue=5.040579004521124e-06)

In [21]:
from scipy.stats import ttest_rel
dataset=dataset.dropna()
st = dataset[dataset['status']=='Not Placed']['degree_p']
st1 = dataset[dataset['status']=='Not Placed']['mba_p']
ttest_rel(st,st1)

Ttest_relResult(statistic=-0.5246500245327441, pvalue=0.6015843363304603)

# 3.Find the reason for non placement from the dataset

# pvalue<0.05 Reject Null Hypothesis
# Ho= There is no significant difference between colunms
# Ha=There is  significant difference between colunms
# Here pvalue are more than 0.05, so accept Null Hypothesis,So there is no significant changes b/w all the columns with respect to salary. 

In [111]:
dataset.corr()

sl_no     ssc_p     hsc_p  degree_p   etest_p     mba_p    salary
sl_no     1.000000 -0.093480 -0.218428 -0.102250  0.041467 -0.072432  0.063764
ssc_p    -0.093480  1.000000  0.293416  0.380657  0.317892  0.430560  0.035330
hsc_p    -0.218428  0.293416  1.000000  0.221307  0.284672  0.329983  0.076819
degree_p -0.102250  0.380657  0.221307  1.000000  0.217683  0.494093 -0.019272
etest_p   0.041467  0.317892  0.284672  0.217683  1.000000  0.284143  0.178307
mba_p    -0.072432  0.430560  0.329983  0.494093  0.284143  1.000000  0.175013
salary    0.063764  0.035330  0.076819 -0.019272  0.178307  0.175013  1.000000

# 4.What kind of relation between salary and mba_p

# [mba_p,salary_p]=0.139823(positive correlation). It means mba_p is directly proportional to the salary column.

In [50]:
dataset.describe()

sl_no       ssc_p       hsc_p    degree_p     etest_p       mba_p  \
count  215.000000  215.000000  215.000000  215.000000  215.000000  215.000000   
mean   108.000000   67.303395   66.334744   66.358558   72.100558   62.278186   
std     62.209324   10.827205   10.586016    7.321524   13.275956    5.833385   
min      1.000000   40.890000   42.750000   50.000000   50.000000   51.210000   
25%     54.500000   60.600000   60.900000   61.000000   60.000000   57.945000   
50%    108.000000   67.000000   65.000000   66.000000   71.000000   62.000000   
75%    161.500000   75.700000   73.000000   72.000000   83.500000   66.255000   
max    215.000000   89.400000   91.150000   88.500000   98.000000   77.890000   

              salary  
count     215.000000  
mean   197615.116279  
std    150305.844169  
min         0.000000  
25%         0.000000  
50%    240000.000000  
75%    282500.000000  
max    706250.000000

In [112]:
from scipy.stats import ttest_ind
dataset=dataset.dropna()
st = dataset[dataset['specialisation']=='Mkt&HR']['salary']
st1 = dataset[dataset['specialisation']=='Mkt&Fin']['salary']
ttest_ind(st, st1)

Ttest_indResult(statistic=-1.790427661842332, pvalue=0.07545741069325959)

# 5.Which specialization is getting minimum salary

# pvalue<0.05 Reject Null Hypothesis
# Ho= There is no significant difference between colunms
# Ha=There is  significant difference between colunms
# Here pvalue is 0.07 so accept Null Hypothesis,So there is no significant changes b/w coloums both specialization Mkt&HR and Mkt&Fin with respect to salary.Both has minimum salary level.

In [68]:
import numpy as np

In [114]:
def count(list1, l, r):
    return len([x for x in list1 if x > l and x <= r]) 

list1=dataset["salary"]
l=500000
r=940000
print("The result is: ",count(list1, l, r))

The result is:  3


# 6.How many of them getting above 500000 salary

# Three students getting salary more than 500000

In [105]:
import scipy.stats as stats
stats.f_oneway(dataset['mba_p'],dataset['etest_p'])

F_onewayResult(statistic=75.35669909200759, pvalue=2.7511196898832345e-16)

# 7.Test the Analysis of Variance between etest_p and mba_p at signifance level 5%.(Make decision using Hypothesis Testing)


# # pvalue<0.05 Reject Null Hypothesis
# Ho= There is no significant difference between colunms
# Ha=There is  significant difference between colunms
# Here pvalue is 2.75, Accept Null Hypothesis,So there is no significant changes b/w  both columns etest_p and mba_p.

In [115]:
from scipy.stats import ttest_ind
dataset=dataset.dropna()
st = dataset[dataset['degree_t']=='Sci&Tech']['salary']
st1 = dataset[dataset['specialisation']=='Mkt&HR']['salary']
ttest_ind(st, st1)

Ttest_indResult(statistic=2.734391160944239, pvalue=0.007496896218767113)

# 8.Test the similarity between the degree_t(Sci&Tech) and specialisation(Mkt&HR) with respect to salary at significance level of 5%.(Make decision using Hypothesis Testing)


# pvalue<0.05 Reject Null Hypothesis
# Ho= There is no significant difference between colunms
# Ha=There is  significant difference between colunms
# Here pvalue is 0.00, Reject Null Hypothesis,So there is significant changes b/w  both columns degree_t(Sci&Tech) and specialisation(Mkt&HR).

In [108]:
from scipy.stats import ttest_rel
dataset=dataset.dropna()
st = dataset[dataset['degree_t']=='Sci&Tech']['etest_p']
st1 = dataset[dataset['degree_t']=='Sci&Tech']['mba_p']
ttest_rel(st,st1)

Ttest_relResult(statistic=4.915474373730152, pvalue=1.5494422054952274e-05)

# 11.Test the similarity between the degree_t(Sci&Tech)with respect to etest_p and mba_p at significance level of 5%.(Make decision using Hypothesis Testing)


# # pvalue<0.05 Reject Null Hypothesis
# Ho= There is no significant difference between colunms
# Ha=There is  significant difference between colunms
# Here pvalue is 1.54, so accept Null Hypothesis,So there is no significant changes b/w coloums both degree_t(Sci&Tech)with respect to etest_p and mba_p.

In [110]:
dataset.corr()

sl_no     ssc_p     hsc_p  degree_p   etest_p     mba_p    salary
sl_no     1.000000 -0.093480 -0.218428 -0.102250  0.041467 -0.072432  0.063764
ssc_p    -0.093480  1.000000  0.293416  0.380657  0.317892  0.430560  0.035330
hsc_p    -0.218428  0.293416  1.000000  0.221307  0.284672  0.329983  0.076819
degree_p -0.102250  0.380657  0.221307  1.000000  0.217683  0.494093 -0.019272
etest_p   0.041467  0.317892  0.284672  0.217683  1.000000  0.284143  0.178307
mba_p    -0.072432  0.430560  0.329983  0.494093  0.284143  1.000000  0.175013
salary    0.063764  0.035330  0.076819 -0.019272  0.178307  0.175013  1.000000

# 12.Which parameter is highly correlated with salary?

# Nearly 90 and above we consider as a highly correlated, but here no one column is highly correlated to salary column.